common/src/main/java/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -460,6 +460,17 @@ class Backend(
         httpClient.clearCaches()
     }
 
+    private fun calculatePostReceiptErrorType(
+        responseCode: Int,
+        purchasesError: PurchasesError
+    ) = if (RCHTTPStatusCodes.isServerError(responseCode)) {
+        PostReceiptErrorType.SERVER_ERROR
+    } else if (purchasesError.code != PurchasesErrorCode.UnsupportedError) {
+        PostReceiptErrorType.CAN_BE_CONSUMED
+    } else {
+        PostReceiptErrorType.CANNOT_BE_CONSUMED
```

Comment:
Well, it's actually the opposite right? As in, if it's an `UnsupportedError` we won't consume. If it's a different error, we will consume. It can be a bit confusing but that's the logic that we have right now though... Will just move things to avoid that negation though so it's a bit clearer